openquakeplatform/openquakeplatform/static/js/irv_viewer_d3_tree.js

```diff
@@ -64,6 +64,51 @@
         var diagonal = d3.svg.diagonal()
             .projection(function(d) { return [d.y, d.x]; });
 
+        function isComputable(node) {
+            if (node.name === 'IRI') {
+                if (typeof node.children === 'undefined') {
+                    return false;
+                }
+                // check if both RI and SVI are computable
+                var areRiAndSviComputable = true;
+                for (var i = 0; i < node.children.length; i++) {
+                    if (!isComputable(node.children[i])) {
+                        areRiAndSviComputable = false;
+                    }
+                }
+                if (areRiAndSviComputable) {
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+            if (node.name === 'SVI') {
+                if (typeof node.children === 'undefined') {
+                    return false;
+                }
+                // Check if all themes are computable
+                var areAllThemesComputable = true;
+                for (var i = 0; i < node.children.length; i++) {
+                    if (!isComputable(node.children[i])) {
+                        areAllThemesComputable = false;
+                    }
+                }
+                if (areAllThemesComputable) {
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+            if (node.name === 'RI' || node.name === 'SVI' ) {
```

Comment:
When you check if SVI is computable, at line 92 you also check if all its themes are computable. So you call the function passing a theme as parameter, and you need the suggested condition to check the computability of that theme.